samcli/commands/list/endpoints/endpoints_context.py

```diff
@@ -18,7 +18,13 @@ class EndpointsContext(ListContext):
     """
 
     def __init__(
-        self, stack_name: str, output: str, region: Optional[str], profile: Optional[str], template_file: Optional[str]
+        self,
+        stack_name: str,
+        output: str,
+        region: Optional[str],
+        profile: Optional[str],
+        template_file: Optional[str],
+        parameter_overrides: dict = {},
```

Comment:
Python has a weird behaviour when it comes to default collections. Read for more info: https://rules.sonarsource.com/python/RSPEC-5717

So it is better to define the default as `None` here and during assignment, you can do some thing like;
```
self.parameter_overrides = parameter_overrides or {}
```

Reply:
TIL, will make that change